app/assets/javascripts/test/model/flycam_accessors.spec.js

```diff
@@ -67,3 +68,45 @@ test("Flycam Accessors should calculate the texture scaling factor (2/2)", t =>
   const texturePosition = accessors.getTextureScalingFactor(state);
   t.deepEqual(texturePosition, 0.5375);
 });
+
+test.only("Flycam Accessors should calculate appropriate zoom factors for datasets with many magnifications.", t => {
+  const scale = [4, 4, 35];
+  const resolutions = [
+    [1, 1, 1],
+    [2, 2, 1],
+    [4, 4, 1],
+    [8, 8, 1],
+    [16, 16, 2],
+    [32, 32, 4],
+    [64, 64, 8],
+    [128, 128, 16],
+    [256, 256, 32],
+    [512, 512, 64],
+    [1024, 1024, 128],
+    [2048, 2048, 256],
+    [4096, 4096, 512],
+  ];
+
+  const maximumZoomPerResolution = _getMaximumZoomForAllResolutions(scale, resolutions);
+
+  // If this test case should fail at some point, the following values may be updated appropriately
+  // to make it pass again. However, it should be validated that zooming out works as expected for
+  // datasets with many magnifications (> 12). Small variations in these numbers shouldn't matter much.
+  const expectedZoomValues = [
```

Comment:
For comparison: before this PR, the function returned

```
[
    1.6000000000000005,
    2.9000000000000017,
    5.199999999999998,
    8.199999999999989,
    16.399999999999967,
    32.800000000000196,
    40.0000000000003,
    40.0000000000003,
    40.0000000000003,
    40.0000000000003,
    40.0000000000003,
    40.0000000000003,
    40.0000000000003,
]
```

(Before yesterdays PR, the values did only increase until 20).